analyzer/evmabibackfill/evm_abi_backfill.go

```diff
@@ -167,134 +168,102 @@ func cleanTxRevertReason(raw string) ([]byte, error) {
 	return base64.StdEncoding.DecodeString(s)
 }
 
+func (p *processor) processEvent(ev *abiEncodedEvent, contractAbi abi.ABI) (*string, []*abiEncodedArg, *ethCommon.Hash) {
+	abiEvent, abiEventArgs, err := abiparse.ParseEvent(ev.EventBody.Topics, ev.EventBody.Data, &contractAbi)
+	if err != nil {
+		p.logger.Warn("error processing event using abi", "err", err)
+		return nil, nil, nil
+	}
+	eventArgs, err := marshalArgs(abiEvent.Inputs, abiEventArgs)
+	if err != nil {
+		p.logger.Warn("error processing event args using abi", "err", err)
+		return nil, nil, nil
+	}
+
+	return &abiEvent.Name, eventArgs, &abiEvent.ID
+}
+
+func (p *processor) processTxCall(tx *abiEncodedTx, contractAbi abi.ABI) (*string, []*abiEncodedArg) {
+	method, abiTxArgs, err := abiparse.ParseData(tx.TxData, &contractAbi)
+	if err != nil {
+		p.logger.Warn("error processing tx using abi", "err", err)
+		return nil, nil
+	}
+	txArgs, err := marshalArgs(method.Inputs, abiTxArgs)
+	if err != nil {
+		p.logger.Warn("error processing tx args using abi", "err", err)
+		return nil, nil
+	}
+
+	return &method.RawName, txArgs
+}
+
+func (p *processor) ProcessTxErr(tx *abiEncodedTx, contractAbi abi.ABI) (*string, []*abiEncodedArg) {
+	var abiErrMsg string
+	var abiErr *abi.Error
+	var abiErrArgs []interface{}
+	var errArgs []*abiEncodedArg
+	if tx.TxRevertReason != nil {
+		txrr, err := cleanTxRevertReason(*tx.TxRevertReason)
+		if err != nil {
+			// This is most likely an older tx with a plaintext revert reason, such
+			// as "reverted: Ownable: caller is not the owner". In this case, we do
+			// not parse the error with the abi.
+			p.logger.Info("encountered likely old-style reverted transaction", "revert reason", tx.TxRevertReason, "tx hash", tx.TxHash, "err", err)
+			return nil, nil
+		}
+		abiErr, abiErrArgs, err = abiparse.ParseError(txrr, &contractAbi)
+		if err != nil || abiErr == nil {
+			p.logger.Warn("error processing tx error using abi", "contract address", "err", err)
+			return nil, nil
+		}
+		abiErrMsg = runtime.TxRevertErrPrefix + abiErr.Name + prettyPrintArgs(abiErrArgs)
+		errArgs, err = marshalArgs(abiErr.Inputs, abiErrArgs)
+		if err != nil {
+			p.logger.Warn("error processing tx error args", "err", err)
+			return nil, nil
+		}
+	}
+
+	return &abiErrMsg, errArgs
+}
+
 func (p *processor) ProcessItem(ctx context.Context, batch *storage.QueryBatch, item *abiEncodedItem) error {
 	// Unmarshal abi
 	contractAbi, err := abi.JSON(bytes.NewReader(item.Abi))
 	if err != nil {
 		return fmt.Errorf("error unmarshalling abi: %w", err)
 	}
 	// Parse data
+	p.logger.Debug("processing item using abi", "contract_address", item.ContractAddr)
 	if item.Event != nil { //nolint:nestif // has complex nested blocks (complexity: 12) (nestif)
```

Comment:
ohhh you know, as I was writing the comment in #610 I was thinking "and I wonder if the parsing decomposes into helper funcs really nicely, or if there's shared state I'm missing". Glad it works out and you factored these out, looks sleek!

And the linter says we don't need the `nolint` any more ;)